docs/spark/how-to-guides/batch-processing.md

```diff
@@ -0,0 +1,124 @@
+ms.custom: mvc,how-to
+---
+
+# How to do batch processing with .NET for Apache Spark
+
+This how-to shows you how to do batch processing using .NET for Apache Spark. If this is your first time using .NET for Apache Spark, check out the [Get started with .NET for Apache Spark](../tutorials/get-started.md) tutorial to learn how to prepare your environment and run your first .NET for Apache Spark application.
+
+## What is batch processing?
+
+Batch processing is the transformation of data at rest, meaning that the source data has already been loaded into data storage. Batch processing is generally performed over large, flat datasets that need to be prepared for further analysis. Log processing and data warehousing are common batch processing scenarios. 
+
+## GitHub projects data example
+
+The following steps demonstrate batch processing of GitHub projects data. 
+
+[GHTorrent](http://ghtorrent.org/) monitors all public GitHub events, such as info about projects, commits, and watchers, and stores the events and their structure in databases. Data collected over different time periods is available as downloadable archives. Because the dump files are very large, this guide uses a [truncated version of the dump file]() that can be downloaded from GitHub.
+
+### Problem
+
+The goal is to analyze information about GitHub projects. First, the data is prepped by removing null data and unnecessary columns. Then, further analysis is performed to discover the number of times different projects have been forked and how recently projects have been updated.
+
+### Solution
+
+The following steps walk you through how to process the GitHub projects data. You can see the [full solution](https://github.com/dotnet/spark/blob/master/examples/Microsoft.Spark.CSharp.Examples/Sql/Batch/GitHubProjects.cs) on GitHub.
+
+1. Create a new .NET console app and add the Microsft.Spark NuGet package. Then, add the following using statements to your code.
+
+   ```csharp
+   using System;
+   using Microsoft.Spark.Sql;
+   using static Microsoft.Spark.Sql.Functions;
+   ```
+
+1. Add the following code to create a `s_referenceDate` variable and a `Run` function. The `Run` function takes the CSV file path as an argument.
+
+   ```csharp
+   // For later use when filtering based on date
+   static readonly DateTime s_referenceDate = new DateTime(2015 10, 20);
+
+   public void Run(string[] args)
+   {
+       if (args.Length != 1)
+       {
+            Console.Error.WriteLine(
+            "Usage: GitHubProjects <path to projects.csv>");
+            Environment.Exit(1);
+        }
+    }
+   ```
+
+1. Inside the `Run` function, add the following code to establish a new SparkSession. The SparkSession is the entry point to programming Spark with the Dataset and DataFrame API. By calling the `spark` object, you can access Spark and DataFrame functionality throughout your program.
+
+   ```csharp
+   SparkSession spark = SparkSession
+                .Builder()
```

Comment:
Seems like there is extra indentation for this and next two lines (compared to indentation for code snippets like the DataFrame snippet in the next section)?